npm/vue/src/index.ts

```diff
@@ -119,17 +273,34 @@ export function mount<Props = any> (
   })
 }
 
+function getComponentDisplayName (componentOptions: any): string {
+  if (componentOptions.name) {
+    return componentOptions.name
+  }
+
+  if (componentOptions.__file) {
+    const filepathSplit = componentOptions.__file.split('/')
+    const fileName = filepathSplit[filepathSplit.length - 1]
+
+    return fileName.replace(/\....?$/, '')
```

Comment:
What does this do? Can you put a comment explaning (very hard to understand regexp without knowing expected input)

Reply:
yep will do

Reply:
Done